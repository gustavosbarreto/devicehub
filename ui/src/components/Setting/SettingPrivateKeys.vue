<template>
  <v-container>
    <v-row align="center" justify="center">
      <v-col sm="8">
        <v-alert
          class="text-subtitle-2"
          type="info"
          title="ShellHub does not save your private keys"
          text="Your private keys remain confidential and are not shared with the ShellHub server.
            Instead, they are securely stored in your browser's local storage."
          color="primary"
          variant="tonal"
          data-test="privacy-policy-alert"
        />
      </v-col>
    </v-row>
    <v-row align="center" justify="center" class="mt-4">
      <v-col sm="8">
        <v-card class="pb-0 elevation-0 bg-v-theme-surface">
          <v-row>
            <v-col>
              <h3>Private Keys</h3>
            </v-col>
            <v-col>
              <div class="d-flex pa-0 align-center bg-v-theme-surface">
                <v-spacer />
                <v-spacer />

                <span>
                  <PrivateKeyAdd data-test="privateKeyFormDialogFirst-component" />
                </span>
              </div>
            </v-col>
          </v-row>

          <PrivateKeyList />
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
import PrivateKeyAdd from "../PrivateKeys/PrivateKeyAdd.vue";
import PrivateKeyList from "../PrivateKeys/PrivateKeyList.vue";
</script>
