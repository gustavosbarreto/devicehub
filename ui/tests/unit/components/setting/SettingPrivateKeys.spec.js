import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import SettingPrivateKeys from '@/components/setting/SettingPrivateKeys';

describe('SettingPrivateKeys', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let wrapper;

  const numberPrivateKeys = 2;

  const privateKeys = [
    {
      name: 'shellhub',
      data: 'BBGVvbmF',
    },
    {
      name: 'shellhub',
      data: 'AbGVvbmF',
    },
  ];

  const headers = [
    {
      text: 'Name',
      value: 'name',
      align: 'center',
    },
    {
      text: 'Fingerprint',
      value: 'data',
      align: 'center',
    },
    {
      text: 'Actions',
      value: 'actions',
      align: 'center',
    },
  ];

  const store = new Vuex.Store({
    namespaced: true,
    state: {
      privateKeys,
      numberPrivateKeys,
    },
    getters: {
      'privatekeys/list': (state) => state.privateKeys,
      'privatekeys/getNumberPrivateKeys': (state) => state.numberPrivateKeys,
    },
    actions: {
    },
  });

  ///////
  // In this case, the first time the user enters the private keys
  // tab, a dialog appears with a certain message.
  ///////

  describe('Dialog is true', () => {
    beforeEach(() => {
      wrapper = shallowMount(SettingPrivateKeys, {
        store,
        localVue,
        stubs: ['fragment'],
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Compare data with default value', () => {
      expect(wrapper.vm.dialog).toEqual(true);
      expect(wrapper.vm.privatekeyPrivacyPolicy).toEqual(false);
      expect(wrapper.vm.headers).toEqual(headers);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.getListPrivateKeys).toEqual(privateKeys);
      expect(wrapper.vm.getNumberPrivateKeys).toEqual(numberPrivateKeys);
    });

    //////
    // HTML validation
    //////

    it('Renders the template with data', () => {
      expect(wrapper.find('[data-test="gotIt-btn"]').exists()).toBe(true);
    });
  });

  ///////
  // In this case, verify data in data table.
  ///////

  describe('Dialog is false', () => {
    beforeEach(() => {
      wrapper = shallowMount(SettingPrivateKeys, {
        store,
        localVue,
        stubs: ['fragment'],
      });

      wrapper.setData({ dialog: false });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Compare data with default value', () => {
      expect(wrapper.vm.dialog).toEqual(false);
      expect(wrapper.vm.privatekeyPrivacyPolicy).toEqual(false);
      expect(wrapper.vm.headers).toEqual(headers);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.getListPrivateKeys).toEqual(privateKeys);
      expect(wrapper.vm.getNumberPrivateKeys).toEqual(numberPrivateKeys);
    });

    //////
    // HTML validation
    //////

    it('Renders the template with components', () => {
      expect(wrapper.find('[data-test="privateKeyFormDialogFirst-component"]').exists()).toBe(true);
    });
    it('Renders the template with data', () => {
      const dt = wrapper.find('[data-test="dataTable-field"]');
      const dataTableProps = dt.vm.$options.propsData;

      expect(dataTableProps.items).toHaveLength(numberPrivateKeys);
    });
  });
});
