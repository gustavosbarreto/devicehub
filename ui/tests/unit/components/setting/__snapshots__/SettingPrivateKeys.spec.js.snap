// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingPrivateKeys Dialog is false Renders the component 1`] = `
<v-form-stub>
  <v-container-stub tag="div">
    <v-row-stub tag="div" align="center" justify="center" class="mt-4 mb-4">
      <v-col-stub sm="8" tag="div">
        <v-card-stub loaderheight="4" tag="div" class="pb-0 elevation-0">
          <div class="d-flex pa-0 align-center">
            <v-spacer-stub tag="div"></v-spacer-stub>
            <v-spacer-stub tag="div"></v-spacer-stub> <span><privatekeyformdialog-stub keyobject="[object Object]" createkey="true" action="private" data-test="privateKeyFormDialogFirst-component"></privatekeyformdialog-stub></span>
          </div>
          <v-data-table-stub mobilebreakpoint="600" items="[object Object],[object Object]" options="[object Object]" sortby="" sortdesc="" customsort="[Function]" page="1" itemsperpage="10" groupby="" groupdesc="" customgroup="[Function]" locale="en-US" customfilter="[Function]" serveritemslength="2" itemkey="id" value="" expanded="" noresultstext="$vuetify.dataIterator.noResultsText" nodatatext="$vuetify.noDataText" loadingtext="$vuetify.dataIterator.loadingText" hidedefaultfooter="true" selectablekey="isSelectable" loaderheight="4" headers="[object Object],[object Object],[object Object]" expandicon="$expand" itemclass="" data-test="dataTable-field"></v-data-table-stub>
        </v-card-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
  <v-dialog-stub opendelay="0" closedelay="0" contentclass="" origin="center center" persistent="true" retainfocus="true" transition="dialog-transition" width="500">
    <v-card-stub loaderheight="4" tag="div">
      <v-card-title-stub tag="div" class="headline grey lighten-2">
        Privacy Policy
      </v-card-title-stub>
      <v-card-text-stub tag="div" class="mt-4">
        The private key is never submitted to ShellHub, it gets stored in your browser’s
        local storage, only the public key gets uploaded and stored by ShellHub.
      </v-card-text-stub>
      <v-divider-stub></v-divider-stub>
      <v-card-actions-stub tag="div" class="px-6">
        <v-checkbox-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Never show this again" ripple="true" valuecomparator="[Function]" indeterminateicon="$checkboxIndeterminate" officon="$checkboxOff" onicon="$checkboxOn"></v-checkbox-stub>
        <v-spacer-stub tag="div"></v-spacer-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button" data-test="gotIt-btn">
          Got it
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
</v-form-stub>
`;

exports[`SettingPrivateKeys Dialog is true Renders the component 1`] = `
<v-form-stub>
  <v-container-stub tag="div">
    <v-row-stub tag="div" align="center" justify="center" class="mt-4 mb-4">
      <v-col-stub sm="8" tag="div">
        <v-card-stub loaderheight="4" tag="div" class="pb-0 elevation-0">
          <div class="d-flex pa-0 align-center">
            <v-spacer-stub tag="div"></v-spacer-stub>
            <v-spacer-stub tag="div"></v-spacer-stub> <span><privatekeyformdialog-stub keyobject="[object Object]" createkey="true" action="private" data-test="privateKeyFormDialogFirst-component"></privatekeyformdialog-stub></span>
          </div>
          <v-data-table-stub mobilebreakpoint="600" items="[object Object],[object Object]" options="[object Object]" sortby="" sortdesc="" customsort="[Function]" page="1" itemsperpage="10" groupby="" groupdesc="" customgroup="[Function]" locale="en-US" customfilter="[Function]" serveritemslength="2" itemkey="id" value="" expanded="" noresultstext="$vuetify.dataIterator.noResultsText" nodatatext="$vuetify.noDataText" loadingtext="$vuetify.dataIterator.loadingText" hidedefaultfooter="true" selectablekey="isSelectable" loaderheight="4" headers="[object Object],[object Object],[object Object]" expandicon="$expand" itemclass="" data-test="dataTable-field"></v-data-table-stub>
        </v-card-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
  <v-dialog-stub opendelay="0" closedelay="0" value="true" contentclass="" origin="center center" persistent="true" retainfocus="true" transition="dialog-transition" width="500">
    <v-card-stub loaderheight="4" tag="div">
      <v-card-title-stub tag="div" class="headline grey lighten-2">
        Privacy Policy
      </v-card-title-stub>
      <v-card-text-stub tag="div" class="mt-4">
        The private key is never submitted to ShellHub, it gets stored in your browser’s
        local storage, only the public key gets uploaded and stored by ShellHub.
      </v-card-text-stub>
      <v-divider-stub></v-divider-stub>
      <v-card-actions-stub tag="div" class="px-6">
        <v-checkbox-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Never show this again" ripple="true" valuecomparator="[Function]" indeterminateicon="$checkboxIndeterminate" officon="$checkboxOff" onicon="$checkboxOn"></v-checkbox-stub>
        <v-spacer-stub tag="div"></v-spacer-stub>
        <v-btn-stub color="primary" tag="button" activeclass="" text="true" type="button" data-test="gotIt-btn">
          Got it
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
</v-form-stub>
`;
